File: schemas/project-management/drizzle/project_statuses.ts
// project_statuses: workflow statuses scoped to a project with ordering and category.
// See README.md for full design rationale.

import { pgTable, pgEnum, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";
import { projects } from "./projects";

export const projectStatusCategoryEnum = pgEnum("project_status_category", [
  "backlog",
  "unstarted",
  "started",
  "completed",
  "cancelled",
]);

export const projectStatuses = pgTable(
  "project_statuses",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    projectId: uuid("project_id").notNull().references(() => projects.id, { onDelete: "cascade" }),
    name: text("name").notNull(),
    color: text("color"),
    category: projectStatusCategoryEnum("category").notNull(),
    position: integer("position").notNull().default(0),
    isDefault: boolean("is_default").notNull().default(false),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_project_statuses_project_id_position").on(table.projectId, table.position),
    index("idx_project_statuses_project_id_category").on(table.projectId, table.category),
  ]
);

File: schemas/project-management/spacetimedb/project_statuses.rs
// project_statuses: Workflow status definitions for tasks within a project.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[derive(SpacetimeType, Clone)]
pub enum ProjectStatusCategory {
    Backlog, // type: String
    Unstarted,
    Started,
    Completed,
    Cancelled,
}

#[spacetimedb::table(name = project_statuses, public)]
pub struct ProjectStatus {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub project_id: String, // UUID — FK → projects.id (cascade delete)
    pub name: String,
    pub color: Option<String>,
    pub category: ProjectStatusCategory,
    pub position: i32,
    pub is_default: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
    // Composite index: (project_id, position)
    // Composite index: (project_id, category)
}
